ansible_base/activitystream/models/entry.py

```diff
@@ -42,6 +45,44 @@ class Meta:
     def __str__(self):
         return f'[{self.created}] {self.get_operation_display()} by {self.created_by}: {self.content_type} {self.object_id}'
 
+    @functools.cached_property
+    def changed_fk_fields(self):
+        """
+        :return: A dictionary of {field_name: pk} for any ForeignKey fields that have changed.
+            the pk is the new pk value, for changed fields.
+        """
+        changed_fks = {}
+
+        if self.changes is None:
+            return changed_fks
+
+        for op in ('added_fields', 'changed_fields', 'removed_fields'):
+            for field_name, value in self.changes.get(op, {}).items():
+                field = self.content_type.model_class()._meta.get_field(field_name)
+                if isinstance(field, models.ForeignKey):
+                    if op == 'changed_fields':
+                        pk = value[1]
+                    else:
+                        pk = value
+                    changed_fks[field_name] = pk
+
+        return changed_fks
+
+    @functools.cached_property
+    def content_object_with_prefetched_changed_fields(self):
```

Comment:
Says "prefetch" in property name, but method does `select_related`, not `prefetch_related`. Maybe saying "cached" might be more clear?

Reply:
Done